resolver-dns/src/main/java/io/netty/resolver/dns/DnsServerAddresses.java

```diff
@@ -16,84 +16,36 @@
 
+import static io.netty.resolver.dns.DefaultDnsServerAddressStreamProvider.defaultAddressArray;
+
 /**
  * Provides an infinite sequence of DNS server addresses to {@link DnsNameResolver}.
  */
 @UnstableApi
 @SuppressWarnings("IteratorNextCanNotThrowNoSuchElementException")
 public abstract class DnsServerAddresses {
-
-    private static final InternalLogger logger = InternalLoggerFactory.getInstance(DnsServerAddresses.class);
-
-    private static final List<InetSocketAddress> DEFAULT_NAME_SERVER_LIST;
-    private static final InetSocketAddress[] DEFAULT_NAME_SERVER_ARRAY;
-    private static final DnsServerAddresses DEFAULT_NAME_SERVERS;
-    static final int DNS_PORT = 53;
-
-    static {
-        final List<InetSocketAddress> defaultNameServers = new ArrayList<InetSocketAddress>(2);
-        try {
-            Class<?> configClass = Class.forName("sun.net.dns.ResolverConfiguration");
-            Method open = configClass.getMethod("open");
-            Method nameservers = configClass.getMethod("nameservers");
-            Object instance = open.invoke(null);
-
-            @SuppressWarnings("unchecked")
-            final List<String> list = (List<String>) nameservers.invoke(instance);
-            for (String a: list) {
-                if (a != null) {
-                    defaultNameServers.add(new InetSocketAddress(SocketUtils.addressByName(a), DNS_PORT));
-                }
-            }
-        } catch (Exception ignore) {
-            // Failed to get the system name server list.
-            // Will add the default name servers afterwards.
-        }
-
-        if (!defaultNameServers.isEmpty()) {
-            if (logger.isDebugEnabled()) {
-                logger.debug(
-                        "Default DNS servers: {} (sun.net.dns.ResolverConfiguration)", defaultNameServers);
-            }
-        } else {
-            Collections.addAll(
-                    defaultNameServers,
-                    SocketUtils.socketAddress("8.8.8.8", DNS_PORT),
-                    SocketUtils.socketAddress("8.8.4.4", DNS_PORT));
-
-            if (logger.isWarnEnabled()) {
-                logger.warn(
-                        "Default DNS servers: {} (Google Public DNS as a fallback)", defaultNameServers);
-            }
-        }
-
-        DEFAULT_NAME_SERVER_LIST = Collections.unmodifiableList(defaultNameServers);
-        DEFAULT_NAME_SERVER_ARRAY = defaultNameServers.toArray(new InetSocketAddress[defaultNameServers.size()]);
-        DEFAULT_NAME_SERVERS = sequential(DEFAULT_NAME_SERVER_ARRAY);
-    }
-
     /**
+     * @deprecated Use {@link DefaultDnsServerAddressStreamProvider#defaultAddressList()}.
+     * <p>
      * Returns the list of the system DNS server addresses. If it failed to retrieve the list of the system DNS server
      * addresses from the environment, it will return {@code "8.8.8.8"} and {@code "8.8.4.4"}, the addresses of the
      * Google public DNS servers.
      */
+    @Deprecated
```

Comment:
@johnou - It is an option to remove these methods completely. I am not opposed to this but some folks prefer a more conservative approach. The behavior also hasn't changed here so there is no "harm" in leaving the methods and cleaning up later.

@normanmaurer @trustin - WDYT? Remove them or leave them deprecated?